### PR TITLE
[4.0] Merge core and com_media builds

### PR DESCRIPTION
**Pull request by C-Lodder**
Pull Request for Issue # .

### Summary of Changes

`com_media` has it's own `package.json` and build scripts. This PR merges them into core. They're now in a single location without any version mismatches.

Some of the dependencies needs to be bumped up

### Testing Instructions

1. Run `npm i` and ensure there are no errors
2. Look at the Media Manager in the backend. Ensure there are no console errors and the page displays properly.

### CODE REVIEW

**Comment by dgrammatiko** [comment]
@wilsonge this is a good starting point, merge it

**Comment by brianteeman** [comment]
```

26% building 135/136 modules 1 active ...joomla-cms\node_modules\sass-loader\lib\loader.js!C:\htdocs\joomla-cms\administrator\components\com_media\resources\styles\mediamanager.scss (node:50588) DeprecationWarning: loaderUtils.parseQuery() received a non-string value which can be problematic, see https://github.com/webpack/loader-utils/issues/56
parseQuery() will be replaced with getOptions() in the next major version of loader-utils.
```

Did not get this message before 

Seems to work ok though as its just a warning

**Comment by C-Lodder** [comment]
@brianteeman dependencies need updating.

**Comment by brianteeman** [comment]
I ran npm update and then npm i again - same issue

or did I misunderstand

**Comment by C-Lodder** [comment]
No sorry. I mean the dependencies need a proper update in the `package.json`. I've updated the bare minimum to get this merge working.

File: webpack.config.js [hunk 1]
```diff
@@ -1,17 +1,40 @@
...
     ],
     output: {
-        path: path.resolve(__dirname, './../../../media/com_media/js'),
+        path: path.resolve(__dirname, './media/com_media/js'),
         publicPath: '/',
         filename: 'mediamanager.min.js'
     },
+    optimization: {
+        minimizer: [
+            new TerserPlugin({
```
Comment (wilsonge) [inline webpack.config.js]: I'd still like to not minimize this in development mode - makes life much harder for debugging. It's just a flag in this plugin so can you implement it please

Comment (C-Lodder) [reply webpack.config.js]: sure, will do

**Comment by C-Lodder** [comment]
@wilsonge `development` mode added in. Remove Terser plugin as webpack has it's own minification process without the need for plugins.

**Comment by wilsonge** [comment]
You added Terser mate. https://github.com/joomla/joomla-cms/pull/28024/files#diff-11e9f7f953edc64ba14b0cc350ae7b9dL73 we were using the built in webpack uglify to minimize before your change

File: webpack.config.js [hunk 2]
```diff
@@ -1,14 +1,16 @@
-var path = require('path');
-var webpack = require('webpack');
-var ExtractTextPlugin = require('extract-text-webpack-plugin');
+const path = require('path');
+const webpack = require('webpack');
+const MiniCssExtractPlugin = require('mini-css-extract-plugin');
+const VueLoaderPlugin = require('vue-loader/lib/plugin')
 
 module.exports = {
+    mode: 'development',
```
Comment (wilsonge) [inline webpack.config.js]: This looks wrong? It's not supposed to be hardcoded. It comes from the `NODE_ENV=development` or `NODE_ENV=production` in the package.json jobs

Comment (C-Lodder) [reply webpack.config.js]: That's the default value. Check the bottom of the file. The `mode` gets changed.

Comment (wilsonge) [reply webpack.config.js]: But if you leave it out entirely and just run the jobs doesn't it bubble through this way by default?

Comment (dgrammatiko) [reply webpack.config.js]: Just use a ternary there as you do in the other places
```js
mode: JSON.stringify(process.env.NODE_ENV) === 'production' ? 'production' : 'development',
```

Comment (wilsonge) [reply webpack.config.js]: Ahh it doesn't automatically set mode (sad times). @dgrammatiko I don't think that works. `JSON. stringify` would result in you needing to check `'"production"'`. But yeah concept seems about right

Comment (dgrammatiko) [reply webpack.config.js]: 😞 just copied that line form a couple lines below, but you get the idea...

Comment (C-Lodder) [reply webpack.config.js]: Guys, the two tasks to build for dev and production work as expected.

**Comment by C-Lodder** [comment]
@wilsonge I know I added it. Was just specifying that I've removed it completely. Wepack minifies files differently now. It doesn't use Uglify. The script is now using webpack 4, not 3

Please just run the two tasks:

```
$ npm run dev:com_media
$ npm run build:com_media
```

and you'll see the difference in the generated files.

**Comment by C-Lodder** [comment]
FYI this also fixes the ability to run `npm i` on Node 13.x
With the current separation, `node-sass 4.12.x` won't work on NodeJS 12.x

**Comment by wilsonge** [comment]
https://github.com/C-Lodder/joomla-cms/pull/7 merge this and we're good to go i think. Still not quite happy with the quality of the builds (i think babel-loader is minifying each file??) but the webpack itself is/isn't minified correctly in dev/prod builds. But that part can be improved later

**Comment by C-Lodder** [comment]
Will merge once you've confirmed you've tested both build scripts

**Comment by C-Lodder** [comment]
@wilsonge It has been merged already

**Comment by infograf768** [comment]
I have tested this item :white_check_mark: successfully on b3d2f388fffe8ef103f4ea7ac199602215a1228c<br /><br />Media loads OK, no console error.<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/28024">issues.joomla.org/tracker/joomla-cms/28024</a>.</sub>

**Comment by HLeithner** [comment]
Thanks

**Pull request merged**